plasmapy/diagnostics/thomson.py

```diff
@@ -111,12 +129,39 @@ def spectral_density(
 
     electron_vel : `~astropy.units.Quantity`, shape (Ne, 3), optional
         Velocity of each electron component in the rest frame. (convertible to m/s)
+        If set, overrides electron_vdir and electron_speed.
         Defaults to a stationary plasma [0, 0, 0] m/s.
 
+    electron_vdir : np.ndarray, shape (Ne,3), optional
+        Unit vectors describing the velocity of each electron population.
+        Unit vectors will be normalized, but must not contain all zero elements.
+        Setting electron_vel overrides this keyword.
```

Comment:
Why are these defined as two separate arguments?  I would suspect you could calculate `electron_vdir` directly from  `electron_vel`. (Same goes for the ion pair.)

Reply:
You definitely can: you can either set `electron_vel` or `electron_vdir` and `electron_speed`. Most of the time, just specifying `electron_vel` is more natural. However, the other mode is necessary for fitting. Consider a plasma with three electron (or ion) populations, each with a different flow velocity. There is not enough information in the spectrum to fit all three velocity components (the spectra only encodes the component of the velocity along the probe beam). Furthermore, in many cases the direction of the flows is known or can be assumed. So, in that case you want to set `electron_vdir` as a fixed parameter but vary `electron_speed`. That can only be done by offering this second way of entering the velocities.